From 495306b5e7b0dd1cfafa2ee577c535e041b648e5 Mon Sep 17 00:00:00 2001
From: Robby Cai <robby.cai@nxp.com>
Date: Mon, 21 Nov 2016 17:08:04 +0800
Subject: [PATCH 1286/5242] MLK-13497 clock: make an accurate pixel clock rate
 for epdc on i.mx6sll

commit  5eee3ac45680b5192ef07e826e251fe12d053003 from
https://source.codeaurora.org/external/imx/linux-imx.git

change parent clock to pll3_pfd2 and calculate out a desired pixel clock
rate. This patch fixed the following warning.
"imx_epdc_v2_fb 20f4000.epdc: Unable to get an accurate EPDC pix clkdesired = 40000000, actual = 63529412"

Signed-off-by: Robby Cai <robby.cai@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/clk/imx/clk-imx6sll.c |    6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/drivers/clk/imx/clk-imx6sll.c b/drivers/clk/imx/clk-imx6sll.c
index 0b1b501..874dab4 100644
--- a/drivers/clk/imx/clk-imx6sll.c
+++ b/drivers/clk/imx/clk-imx6sll.c
@@ -363,5 +363,11 @@ static void __init imx6sll_clocks_init(struct device_node *ccm_node)
 	clk_set_parent(clks[IMX6SLL_CLK_PERIPH], clks[IMX6SLL_CLK_PERIPH_PRE]);
 
 	clk_set_rate(clks[IMX6SLL_CLK_AHB], 132000000);
+
+	/* Configure EPDC clocks */
+	imx_clk_set_rate(clks[IMX6SLL_CLK_PLL3_PFD2], 320000000);
+	clk_set_parent(clks[IMX6SLL_CLK_EPDC_PRE_SEL],
+		clks[IMX6SLL_CLK_PLL3_PFD2]);
+
 }
 CLK_OF_DECLARE_DRIVER(imx6sll, "fsl,imx6sll-ccm", imx6sll_clocks_init);
-- 
1.7.9.5

